lib/docker-toolbelt.coffee

```diff
@@ -107,12 +112,49 @@ EEXIST = code: 'EEXIST'
 ignore = ->
 MIN_PAGE_SIZE = 4096
 
+pathPrefixRemover = (prefix) ->
+	(path) ->
+		slice = path.substr(prefix.length)
+		# return original if path doesn't start with given prefix
+		return if "#{prefix}#{slice}" == path then slice else path
+
+overlay2MountWithDisposer = (fsRoot, target, lowers, diffDir, workDir) ->
+	# If no lower, just return diff directory
+	return Promise.resolve(diffDir) if !lowers
+
+	fs.mkdirAsync(target)
+	.catch EEXIST, ignore
+	.then ->
+		options = "lowerdir=#{lowers},upperdir=#{diffDir},workdir=#{workDir}"
```

Comment:
In general I agree, though I think it's not really worth the effort -- this is not arbitrary user input. I could possibly escape the single quotes but I wouldn't even know where to start trying to escape commas -- should I escape with a backslash, how many of them, how will this propagate down to the overlayFs driver and how will it handle this input, etc. I hear you, but I really don't think it's worth the effort.